src/main/java/com/keshavarzi/airplanestats/model/UserEntity.java

```diff
@@ -0,0 +1,141 @@
+package com.keshavarzi.airplanestats.model;
+
+import jakarta.annotation.Nonnull;
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.ForeignKey;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.JoinTable;
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.util.Collection;
+import java.util.List;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@Entity
+@Table(name = "user", schema = "user_data")
+public class UserEntity implements Cloneable {
+
+    @Id
+    @Nonnull
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", table = "user", unique = true, nullable = false)
+    private Long userId;
+
+    @Nonnull
+    @Column(name = "email", table = "user", unique = true, nullable = false)
+    private String email;
+
+    @Nonnull
+    @Column(name = "password", table = "user", nullable = false)
+    private String password;
+
+    @Getter(AccessLevel.NONE)
+    @Setter(AccessLevel.NONE)
+    @OneToMany(mappedBy = "userEntity", targetEntity = RouteEntity.class, orphanRemoval = true, cascade = CascadeType.ALL)
+    private Collection<RouteEntity> savedRoutes;
+
+    // Exclusion is because associations impact performance
+    @Getter(AccessLevel.NONE)
+    @Setter(AccessLevel.NONE)
+    @OneToMany(mappedBy = "userEntity", targetEntity = Trip.class, orphanRemoval = true, cascade = CascadeType.ALL)
+    private Collection<Trip> savedTrips;
```

Comment:
storing a list of objects from another table can be very heavy especially if you have a user that heavily stores trips. Considering redesigning this portion or have some sort of limit.